AssertionError: wrong color format 'noinherit' when using `$XONSH_COLOR_STYLE = 'bw'`

When I set `$XONSH_COLOR_STYLE = 'bw'` in my `~/.xonshrc`, I get the following error on opening `xonsh`:
```
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/xonsh/__amalgam__.py", line 21338, in main
    sys.exit(main_xonsh(args))
  File "/usr/lib/python3.9/site-packages/xonsh/__amalgam__.py", line 21376, in main_xonsh
    shell.shell.cmdloop()
  File "/usr/lib/python3.9/site-packages/xonsh/ptk_shell/shell.py", line 385, in cmdloop
    line = self.singleline(auto_suggest=auto_suggest)
  File "/usr/lib/python3.9/site-packages/xonsh/ptk_shell/shell.py", line 316, in singleline
    pyghooks.xonsh_style_proxy(self.styler)
  File "/usr/lib/python3.9/site-packages/xonsh/pyghooks.py", line 461, in xonsh_style_proxy
    class XonshStyleProxy(Style):
  File "/home/jelmer/.local/lib/python3.9/site-packages/pygments/style.py", line 112, in __new__
    ndef[4] = colorformat(styledef[3:])
  File "/home/jelmer/.local/lib/python3.9/site-packages/pygments/style.py", line 79, in colorformat
    assert False, "wrong color format %r" % text
AssertionError: wrong color format 'noinherit'
Xonsh encountered an issue during launch
Failback to /usr/bin/zsh
```

Interestingly, I do not run into any problems if I use e.g. `default` and set `$XONSH_COLOR_STYLE = 'bw' in the terminal itself. The colors then update and everything works, but of course when I restart the shell it will use the default color scheme again.

## xonfig
```
+------------------+----------------------+
| xonsh            | 0.10.1               |
| Git SHA          | e9b12c8b             |
| Commit Date      | Jul 24 22:47:37 2021 |
| Python           | 3.9.6                |
| PLY              | 3.11                 |
| have readline    | True                 |
| prompt toolkit   | 3.0.20               |
| shell type       | prompt_toolkit       |
| history backend  | json                 |
| pygments         | 2.10.0               |
| on posix         | True                 |
| on linux         | True                 |
| distro           | manjaro              |
| on wsl           | False                |
| on darwin        | False                |
| on windows       | False                |
| on cygwin        | False                |
| on msys2         | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
| on jupyter       | False                |
| jupyter kernel   | None                 |
| xontrib          | []                   |
+------------------+----------------------+
```

```
# XONSH WEBCONFIG START
$PROMPT = '{env_name}{BOLD_GREEN}{user}@{hostname}{BOLD_BLUE} {cwd}{branch_color}{curr_branch: {}}{RESET} {BOLD_BLUE}{prompt_end}{RESET} '
$XONSH_COLOR_STYLE = 'bw'
# XONSH WEBCONFIG END
```

Potentially relevant pip packages:
```
xonsh                         0.10.1
xonsh-autoxsh                 0.3
xontrib-avox                  0.8
xontrib-histcpy               0.1
xontrib-onepath               0.3.2
xontrib-output-search         0.5.4
xontrib-powerline             0.4.0
xontrib-powerline2            1.5.0
xontrib-prompt-bar            0.4.1
xontrib-ssh-agent             1.0.12
xontrib-z                     0.4
Pygments                      2.10.0
```

Python version: 3.9.6

## Expected Behavior
I'd expect the `bw` scheme to work, since `xonfig colors bw` works fine as well. For some reason, it appears to cause some error only when set from startup.

## Current Behavior
See above

### Traceback (if applicable)
See above

## Steps to Reproduce
See above

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

I get a similar error running `xonfig web` from a shell.

```
$ xonfig web
Web config started at 'http://localhost:8421'. Hit Crtl+C to stop.
ERROR:root:Failed to format Xonsh code AssertionError("wrong color format 'noinherit'"). 'bw'
Traceback (most recent call last):
  File "/Users/tidux/Library/Python/3.9/lib/python/site-packages/xonsh/webconfig/xonsh_data.py", line 194, in render_colors
    display = html_format(token_stream, style=style)
  File "/Users/tidux/Library/Python/3.9/lib/python/site-packages/xonsh/webconfig/xonsh_data.py", line 51, in html_format
    proxy_style = xonsh_style_proxy(XonshStyle(style))
  File "/Users/tidux/Library/Python/3.9/lib/python/site-packages/xonsh/pyghooks.py", line 453, in xonsh_style_proxy
    class XonshStyleProxy(Style):
  File "/Users/tidux/Library/Python/3.9/lib/python/site-packages/pygments/style.py", line 112, in __new__
    ndef[4] = colorformat(styledef[3:])
  File "/Users/tidux/Library/Python/3.9/lib/python/site-packages/pygments/style.py", line 79, in colorformat
    assert False, "wrong color format %r" % text
AssertionError: wrong color format 'noinherit'
ERROR:root:Failed to parse color-display ParseError('syntax error: line 1, column 0'). 'import sys\necho "Welcome $USER on" @(sys.platform)\n\ndef func(x=42):\n    d = {"xonsh": True}\n    return d.get("xonsh") and you\n\n# This is a comment\n![env | uniq | sort | grep PATH]\n'
127.0.0.1 - - [07/May/2022 16:04:57] "GET / HTTP/1.1" 200 -
127.0.0.1 - - [07/May/2022 16:04:57] "GET /js/bootstrap.min.css HTTP/1.1" 304 -
127.0.0.1 - - [07/May/2022 16:04:57] code 404, message File not found
127.0.0.1 - - [07/May/2022 16:04:57] "GET /js/xonsh_sticker.svg HTTP/1.1" 404 -
```

Identical behavior is observed on Darwin/aarch64 and Linux/x86_64 with different terminal emulators, so I'm pretty sure it's not a platform specific issue.

--- a/news/no_background_inherit.rst
+++ b/news/no_background_inherit.rst
@@ -1,0 +1,24 @@
+**Added:**
+
+* <news item>
+
+**Changed:**
+
+* <news item>
+
+**Deprecated:**
+
+* <news item>
+
+**Removed:**
+
+* <news item>
+
+**Fixed:**
+
+* ``pygments`` startup crash when incorrect prepending ``bg:`` to ``noinherit``
+  style directives
+
+**Security:**
+
+* <news item>

--- a/xonsh/pyghooks.py
+++ b/xonsh/pyghooks.py
@@ -177,7 +177,10 @@ def color_name_to_pygments_code(name, styles):
             fgcolor = color
         else:
             fgcolor = styles[getattr(Color, color)]
-        res = "bg:" + fgcolor
+        if fgcolor == "noinherit":
+            res = "noinherit"
+        else:
+            res = f"bg:{fgcolor}"
     else:
         # have regular, non-background color
         mods = parts["modifiers"]

